.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,22 @@
+name: Building & Publishing
+
+on:
+  push:
+    tags: ["v*"]
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+
+  # Release to PyPI after a RELEASE gets PUBLISHED in @spacetelescope/astrocut/main
+  publish-to-pypi:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish_pure_python.yml@v1
+    name: Publish to PyPI
+
+    # We only want to upload to PyPI after a new release has been published on the @spacetelescope/astrocut repo
+    if: github.repository_owner == 'spacetelescope' && github.event_name == 'push'
+    with:
+      test_extras: test
+      test_command: pytest astrocut/tests/test*.py
```

Comment:
This is the command I run in a terminal (from the base directory of my local clone of `astrocut` repo) to get tests to run. I have also seen an example [where warnings are ignored](https://github.com/spacetelescope/acstools/blob/c01ae65bff403d9b5b6649a9f58bd3d1712fdca1/.github/workflows/publish-to-pypi.yml#L39C41-L39C41) but I opted to leave warnings in, in case we want to see if they change etc. Running these tests on my local machine gives me over 200 warnings.

Reply:
perhaps we should make a ticket to address some of these warnings?  if the warnings are not ignored, will the cause the tests to fail?

Reply:
I was wondering the same thing. When testing is commanded, [this action](https://github.com/OpenAstronomy/build-python-dist/blob/main/action.yml) is used, but I looked through that repo and don't see any clues about whether warnings are considered failures. For now I'm assuming that, since our CI builds currently pass even though we get warnings, that running the same tests in this action will pass too?